src/api.ts

```diff
@@ -385,23 +388,37 @@ function createApi({
         tileset.name
       )
 
+      const tilesetImportWorker = new Worker(
+        path.resolve(__dirname, './lib/mbtiles_import_worker.js'),
+        { workerData: { dbPath: db.name } }
+      )
+
+      const importId = generateId()
+
+      activeWorkers.set(importId, tilesetImportWorker)
+
       // TODO: `style` is not guaranteed to exist since the tileset could be a vector tileset
       // and we don't generate a style for those on tileset creation yet.
       // Absence presents various complications when creating and updating offline area and imports in db
       tilesetImportWorker.postMessage({
         type: 'importMbTiles',
-        importId: generateId(),
+        importId,
         mbTilesDbPath: mbTilesDb.name,
         styleId,
         tilesetId,
       })
 
       return new Promise((res, rej) => {
         // TODO: What else has to be called when this occurs? e.g. terminating the import
-        tilesetImportWorker.addListener('error', (err) => {
+        tilesetImportWorker.on('error', (err) => {
+          activeWorkers.delete(importId)
```

Comment:
Do we need to terminate the worker here? And similarly below?

Reply:
yeah probably makes sense to for both cases 👍

Reply:
looking at it more closely, i think we only need to add something for the `error` event case. if the `exit` event is emitted, it's supposed to be the last event emitted by the worker before execution finishes (https://nodejs.org/docs/latest-v12.x/api/worker_threads.html#worker_threads_event_exit)

Reply:
See 0d75ffb3f980d7f4bceefa60691974ba6fd7cdd1 for relevant changes